qiskit/compiler/scheduler.py

```diff
@@ -67,38 +72,38 @@ def schedule(
     """
     arg_circuits_list = isinstance(circuits, list)
     start_time = time()
-    if backend and getattr(backend, "version", 0) > 1:
-        if inst_map is None:
-            inst_map = backend.instruction_schedule_map
-        if meas_map is None:
-            meas_map = backend.meas_map
-        if dt is None:
-            dt = backend.dt
-    else:
-        if inst_map is None:
-            if backend is None:
-                raise QiskitError(
-                    "Must supply either a backend or InstructionScheduleMap for scheduling passes."
-                )
-            defaults = backend.defaults()
+    if target is None:
+        if isinstance(backend, BackendV2):
+            target = backend.target
+            if inst_map:
+                target.update_from_instruction_schedule_map(inst_map=inst_map)
+        elif isinstance(backend, BackendV1):
+            defaults = backend.defaults() if hasattr(backend, "defaults") else None
             if defaults is None:
                 raise QiskitError(
                     "The backend defaults are unavailable. The backend may not support pulse."
                 )
-            inst_map = defaults.instruction_schedule_map
-        if meas_map is None:
-            if backend is None:
+            if backend.configuration() is not None:
+                target = convert_to_target(
+                    configuration=backend.configuration(),
+                    properties=backend.properties(),
+                    defaults=defaults,
+                )
+                if inst_map:
+                    target.update_from_instruction_schedule_map(inst_map=inst_map)
+            else:
+                raise QiskitError("Must specify backend that has a configuration.")
+        else:
+            if meas_map and inst_map:
+                target = Target(concurrent_measurements=meas_map, dt=dt)
+                target.update_from_instruction_schedule_map(inst_map=inst_map)
+            else:
                 raise QiskitError(
-                    "Must supply either a backend or a meas_map for scheduling passes."
+                    "Must specify either target, backend, "
+                    "or both meas_map and inst_map for scheduling passes."
                 )
-            meas_map = backend.configuration().meas_map
-        if dt is None:
-            if backend is not None:
-                dt = backend.configuration().dt
-
-    schedule_config = ScheduleConfig(inst_map=inst_map, meas_map=meas_map, dt=dt)
     circuits = circuits if isinstance(circuits, list) else [circuits]
-    schedules = parallel_map(schedule_circuit, circuits, (schedule_config, method))
+    schedules = parallel_map(schedule_circuit, circuits, (None, target, method))
```

Comment:
`schedule_circuit` is executed by input that schedule_config is `None` and target is `target`.